monkey/infection_monkey/monkey.py

```diff
@@ -140,11 +139,11 @@ def _get_arguments(args):
         return opts
 
     # TODO: By the time we finish 2292, _connect_to_island_api() may not need to return `server`
-    def _connect_to_island_api(self) -> Tuple[str, IIslandAPIClient]:
+    def _connect_to_island_api(self) -> str:
         logger.debug(f"Trying to wake up with servers: {', '.join(self._opts.servers)}")
         server_clients = find_available_island_apis(self._opts.servers)
 
-        server, island_api_client = self._select_server(server_clients)
+        server, self.island_api_client = self._select_server(server_clients)
```

Comment:
Set `self.island_api_client` in `__init__()`. It's easier to understand what attributes the class has if they're all defined/initialized in one place.

Reply:
Make `self.island_api_client` protected with `self._island_api_client`. The dropper does not need access to it.